### ISSUE TITLE
Theia application failed to start when setting `webPreferences` field in package.json 

### ISSUE THREAD

**Issue by Jackson-he** [eclipse-theia/theia#12915]
### Bug Description:
I have created a Theia application, and I want to launch it in Electron. In order to enable Node.js support for the launch window, I need to configure the `theia.frontend.config.electron.windowOptions.webPreferences.nodeIntegration` field in the `package.json`. However, when I configure this field, the application fails to start. If I remove this field, the application starts successfully.

### Steps to Reproduce:
1. Create a demo according to the official steps like: 
   npm install -g yo generator-theia-extension
   mkdir theia-hello-world-extension
   cd theia-hello-world-extension
   yo theia-extension # select the 'Hello World' option and complete the prompts
3. Navigate to the 'electron-app' direcory and add the `theia.frontend.config.electron.windowOptions.webPreferences.nodeIntegration` field to the `package.json`, setting it to either 'true' or 'false'.
4. Then start the electron app, you can see the error like below:
![image](https://github.com/eclipse-theia/theia/assets/26061572/5012b8f0-53dc-4837-9755-dd9fd3fa40a2)

### Additional Information
- Operating System: MacOs Ventura 13.2.1
- Theia Version: 1.41.0

**Comment by msujew** [comment]
I believe the logic to configure our window options is faulty:

https://github.com/eclipse-theia/theia/blob/35865a9c8ba51c0fbd8cf6b9e44f38c6ec882b16/packages/core/src/electron-main/electron-main-application.ts#L302-L320

When the `webPreferences` object is set on the config object, it completely overrides the `webPreferences` of the configured object, including `preload` (which we definitely need).

**Comment by Jackson-he** [comment]
> I believe the logic to configure our window options is faulty:
> 
> https://github.com/eclipse-theia/theia/blob/35865a9c8ba51c0fbd8cf6b9e44f38c6ec882b16/packages/core/src/electron-main/electron-main-application.ts#L302-L320
> 
> When the `webPreferences` object is set on the config object, it completely overrides the `webPreferences` of the configured object, including `preload` (which we definitely need).

yeah, I attempted to fix this issue on my own, then I encountered a new problem that I couldn't use 'require' in the loaded page. However, this wasn't an issue in versions prior to 1.36.0. I set `nodeIntegration ` and `contextIsolation ` to 'true' on the `webPreferences ` object.
